tests/core/recipes/vasp_recipes/mocked/test_vasp_recipes.py

```diff
@@ -19,6 +24,91 @@
 
 DEFAULT_SETTINGS = SETTINGS.model_copy()
 
+def test_nscf_job(tmp_path, monkeypatch, caplog):
+    monkeypatch.chdir(tmp_path)
+
+    atoms = bulk("Al")
+    calc_kwargs = {}
+    kpoints_mode = "uniform"
+    calculate_optics=True
+
+    output = nscf_job(
+        atoms,
+        prev_dir=tmp_path,
+        bandgap=0.0,
+        nbands_factor=1.0,
+        preset="BulkSet",
+        kpoints_mode=kpoints_mode,
+        calculate_optics=calculate_optics,
+        **calc_kwargs,
+    )
+
+    # Assertions on the output dictionary
+    assert "nsites" in output
+    assert "parameters" in output
+    assert "results" in output
+
+    assert output["parameters"]["ismear"] == -5
+    if calculate_optics:
+        assert output["parameters"]["loptics"] is True
+        assert output["parameters"]["lreal"] is False
+        assert output["parameters"]["cshift"] == 1e-5
+    assert output["parameters"]["lorbit"] == 11
+    assert output["parameters"]["lwave"] is False
+    assert output["parameters"]["lcharg"] is False
+    assert output["parameters"]["nsw"] == 0
+    uniform_isym = 2 if kpoints_mode == "uniform" else 0
+    assert output["parameters"]["isym"] == uniform_isym
+    assert output["parameters"]["icharg"] == 11
+    assert output["parameters"].get("kspacing") is None
+
+    # test nbands and log when vasprun.xml doesn't exist
+    vasprun_exists = False
+    if vasprun_exists:
+        assert "nbands" in output["parameters"]
+    else:
+        assert output["parameters"].get("nbands") is None
+        assert "vasprun.xml* file does not exist in the specified directory." in caplog.text
+
+
+    # test nbands when vasprun.xml exists
+    from shutil import copy
+    copy(MOCKED_DIR / "vasprun.xml.gz", tmp_path / "vasprun.xml.gz")
+    output = nscf_job(
+        atoms,
+        prev_dir=tmp_path,
+        nbands_factor=1.2,
+        kpoints_mode="uniform",
+        **calc_kwargs,
+    )
+    assert "nbands" in output["parameters"]
+
+    # test kpoints_mode line mode
+    kpoints_mode = "line"
+    output = nscf_job(
```

Comment:
Note that at this point, the `vasprun.xml.gz` copied from before is still present in the working directory. Since there are some I/O operations being done beforehand, it might be best to split this up into 3 individual test functions (e.g. `test_nscf_job1`, `test_nscf_job2`, etc) to prevent unexpected behavior.